### PR TITLE
Add support for global values

### PR DESCRIPTION
**Pull request by drbild**
## What?

Add a resource and data source for global values.

The presets API has a couple of oddities that don't play nice with Terraform:

## How to handle deletion?

The global values API doesn't currently support deletion.  Terraform requires the resource have a delete method. 

I've implemented it to:
- issue a diagnostic warning that the resource is being removed from the TF state, but not actually deleted.
- remove the resource from the state file
- not delete the resource via the API (obviously)

We could also chose to have the delete function fail (and thus not remove the resource from the state file).  The user would be forced to perform state surgery themselves.  I believe this is a worse experience.

## How to handle immutable options?

Similarly, the options collection is append only. Existing elements cannot be deleted.  My implementation does nothing proactive or special here.  If an option is deleted from the config, the `tf apply` will fail with the error message from the API.

### CODE REVIEW

File: internal/client/global_values.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	body, resp, err := s.Client.Get(ctx, path, url.Values{}, new(GlobalValue))
+	if err != nil {
+		return nil, resp, err
+	}
+	return body.(*GlobalValue), resp, nil
+}
+
+func (s *GlobalValuesService) GetGlobalValueByKey(ctx context.Context, key string) (*GlobalValue, *http.Response, error) {
+	query := url.Values{}
+	query.Set("key", key)
```
Comment (nagl-resourcely) [inline internal/client/global_values.go:73]: =) Ah, the old 'list by id' pattern.

**Review by nagl-resourcely — APPROVED** [review]
Nice, especially the validator.

File: internal/provider/global_value_resource.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+	resp *resource.DeleteResponse,
+) {
+	// Retrieve from state
+	var state *GlobalValueResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Not implemented; the presets API does not support DELETE
```
Comment (gautamr95) [inline internal/provider/global_value_resource.go:295]: ooc was there a reason we didn't build a delete api?

**Review by gautamr95 — APPROVED** [review]

File: internal/provider/global_value_resource.go (hunk 2 above)
Comment (drbild) [reply internal/provider/global_value_resource.go:290]: It's a long story. Let's discuss next week.

**Pull request merged**